### PR TITLE
Support mounting of external secrets

### PR DESCRIPTION
**Pull request by torbsto**

### CODE REVIEW

**Review by philipp94831 — COMMENTED** [review]
Did you test it?

**Comment by torbsto** [comment]
> Did you test it?

Tested :+1:

File: charts/producer-app/templates/pod.yaml [hunk 1]
```diff
@@ -109,18 +109,35 @@ spec:
...
+    {{- if or (.Values.files) (.Values.secretFilesRefs)}}
     volumeMounts:
       {{- range $key, $value := .Values.files }}
       - name: config
         mountPath: {{ printf "%s/%s" $value.mountPath $key | quote }}
         subPath: {{ $key | quote }}
-      {{- end }}
-  {{- end }}
-  {{- if .Values.files }}
+          {{- end }}
```
Comment (philipp94831) [inline charts/producer-app/templates/pod.yaml:118]:
```suggestion
      {{- end }}
```

File: charts/producer-app-cleanup-job/templates/job.yaml [hunk 2]
```diff
@@ -109,18 +109,35 @@ spec:
             - name: JAVA_TOOL_OPTIONS
               value: '-XX:MaxRAMPercentage={{ printf "%.1f" .Values.javaOptions.maxRAMPercentage }}
                 {{ .Values.javaOptions.others | join " " }}'
-        {{- if .Values.files }}
+          {{- if or (.Values.files) (.Values.secretFilesRefs)}}
```
Comment (philipp94831) [inline charts/producer-app-cleanup-job/templates/job.yaml:112]:
```suggestion
          {{- if or (.Values.files) (.Values.secretFilesRefs) }}
```

File: charts/producer-app/templates/pod.yaml [hunk 3]
```diff
@@ -109,18 +109,35 @@ spec:
     readinessProbe:
     {{- .Values.readinessProbe | toYaml | nindent 6 }}
     {{- end }}
-  {{- if .Values.files }}
+    {{- if or (.Values.files) (.Values.secretFilesRefs)}}
```
Comment (philipp94831) [inline charts/producer-app/templates/pod.yaml:112]:
```suggestion
    {{- if or (.Values.files) (.Values.secretFilesRefs) }}
```

File: charts/streams-app-cleanup-job/templates/job.yaml [hunk 4]
```diff
@@ -138,18 +138,35 @@ spec:
             - name: JAVA_TOOL_OPTIONS
               value: '-XX:MaxRAMPercentage={{ printf "%.1f" .Values.javaOptions.maxRAMPercentage }}
                 {{ .Values.javaOptions.others | join " " }}'
-        {{- if .Values.files }}
+         {{- if or (.Values.files) (.Values.secretFilesRefs)}}
```
Comment (philipp94831) [inline charts/streams-app-cleanup-job/templates/job.yaml:141]:
```suggestion
          {{- if or (.Values.files) (.Values.secretFilesRefs) }}
```

File: charts/streams-app-cleanup-job/templates/job.yaml [hunk 5]
```diff
@@ -138,18 +138,35 @@ spec:
...
+              {{- end }}
+              {{- if .subPath}}
+              subPath: {{.subPath }}
+              {{- end }}
+            {{- end}}
         {{- end }}
-      {{- if .Values.files }}
+      {{- if or (.Values.files) (.Values.secretFilesRefs) }}
       volumes:
+      {{- if .Values.files }}
```
Comment (philipp94831) [inline charts/streams-app-cleanup-job/templates/job.yaml:161]:
```suggestion
        {{- if .Values.files }}
```

File: charts/streams-app-cleanup-job/templates/job.yaml [hunk 6]
```diff
@@ -138,18 +138,35 @@ spec:
...
         - name: config
           configMap:
             name: {{ template "streams-app-cleanup-job.name" . }}
       {{- end }}
```
Comment (philipp94831) [inline charts/streams-app-cleanup-job/templates/job.yaml:165]:
```suggestion
        {{- end }}
```

File: charts/producer-app/templates/pod.yaml [hunk 7]
```diff
@@ -109,18 +109,35 @@ spec:
...
     - name: config
       configMap:
-        name: {{ template "producer-app.name" . }}
+        name: {{ template "producer-app.chart.name" . }}
+    {{- end }}
+    {{- range .Values.secretFilesRefs }}
+    - name: {{ .volume }}
+      secret:
+        secretName: {{ .name }}
+    {{- end}}
```
Comment (philipp94831) [inline charts/producer-app/templates/pod.yaml:141]:
```suggestion
    {{- end }}
```

File: charts/producer-app-cleanup-job/templates/job.yaml [hunk 8]
```diff
@@ -109,18 +109,35 @@ spec:
...
+            {{- range .Values.secretFilesRefs }}
+            - name: {{ .volume }}
+              mountPath: {{ .mountPath }}
+              {{- if .readOnly }}
+              readOnly: true
+              {{- end }}
+              {{- if .subPath}}
+              subPath: {{.subPath }}
+              {{- end }}
+            {{- end}}
```
Comment (philipp94831) [inline charts/producer-app-cleanup-job/templates/job.yaml:128]:
```suggestion
            {{- end }}
```

File: charts/producer-app-cleanup-job/templates/job.yaml [hunk 9]
```diff
@@ -109,18 +109,35 @@ spec:
...
+        {{- if .Values.files }}
         - name: config
           configMap:
             name: {{ template "producer-app-cleanup-job.name" . }}
+        {{- end }}
+        {{- range .Values.secretFilesRefs }}
+        - name: {{ .volume }}
+          secret:
+            secretName: {{ .name }}
+        {{- end}}
```
Comment (philipp94831) [inline charts/producer-app-cleanup-job/templates/job.yaml:141]:
```suggestion
        {{- end }}
```

File: charts/producer-app/templates/pod.yaml [hunk 10]
```diff
@@ -109,18 +109,35 @@ spec:
...
+      {{- range .Values.secretFilesRefs }}
+      - name: {{ .volume }}
+        mountPath: {{ .mountPath }}
+        {{- if .readOnly }}
+        readOnly: true
+        {{- end }}
+        {{- if .subPath}}
+        subPath: {{.subPath }}
+        {{- end }}
+      {{- end}}
```
Comment (philipp94831) [inline charts/producer-app/templates/pod.yaml:128]:
```suggestion
      {{- end }}
```

File: charts/streams-app-cleanup-job/templates/job.yaml [hunk 11]
```diff
@@ -138,18 +138,35 @@ spec:
...
+            {{- range .Values.secretFilesRefs }}
+            - name: {{ .volume }}
+              mountPath: {{ .mountPath }}
+              {{- if .readOnly }}
+              readOnly: true
+              {{- end }}
+              {{- if .subPath}}
+              subPath: {{.subPath }}
+              {{- end }}
+            {{- end}}
```
Comment (philipp94831) [inline charts/streams-app-cleanup-job/templates/job.yaml:157]:
```suggestion
            {{- end }}
```

File: charts/streams-app-cleanup-job/templates/job.yaml [hunk 12]
```diff
@@ -138,18 +138,35 @@ spec:
...
+      {{- if .Values.files }}
         - name: config
           configMap:
             name: {{ template "streams-app-cleanup-job.name" . }}
       {{- end }}
+        {{- range .Values.secretFilesRefs }}
+        - name: {{ .volume }}
+          secret:
+            secretName: {{ .name }}
+        {{- end}}
```
Comment (philipp94831) [inline charts/streams-app-cleanup-job/templates/job.yaml:170]:
```suggestion
        {{- end }}
```

File: charts/streams-app/templates/deployment.yaml [hunk 13]
```diff
@@ -178,13 +178,23 @@ spec:
...
+            {{- range .Values.secretFilesRefs }}
+            - name: {{ .volume }}
+              mountPath: {{ .mountPath }}
+              {{- if .readOnly }}
+              readOnly: true
+              {{- end }}
+              {{- if .subPath}}
+              subPath: {{.subPath }}
+              {{- end }}
+            {{- end}}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:197]:
```suggestion
            {{- end }}
```

File: charts/streams-app/templates/deployment.yaml [hunk 14]
```diff
@@ -245,6 +255,11 @@ spec:
           configMap:
             name: {{ template "streams-app.name" . }}
         {{- end }}
+        {{- range .Values.secretFilesRefs }}
+        - name: {{ .volume }}
+          secret:
+            secretName: {{ .name }}
+        {{- end}}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:262]:
```suggestion
        {{- end }}
```

**Review by philipp94831 — CHANGES_REQUESTED** [review]

**Review by philipp94831 — CHANGES_REQUESTED** [review]
Please add documentation to READMEs

**Review by philipp94831 — APPROVED** [review]

**Pull request merged**